include/robot_interfaces/sensors/pylon_driver.hpp

```diff
@@ -116,30 +124,23 @@ class PylonDriver : public SensorDriver<CameraObservation>
         CameraObservation image_frame;
         Pylon::CGrabResultPtr ptr_grab_result;
 
-        try
-        {
-            camera_.RetrieveResult(
-                5000, ptr_grab_result, Pylon::TimeoutHandling_ThrowException);
-            image_frame.time_stamp =
-                real_time_tools::Timer::get_current_time_sec();
+        camera_.RetrieveResult(
+            5000, ptr_grab_result, Pylon::TimeoutHandling_ThrowException);
+        image_frame.time_stamp = real_time_tools::Timer::get_current_time_sec();
 
-            if (ptr_grab_result->GrabSucceeded())
-            {
-                format_converter_.Convert(pylon_image_, ptr_grab_result);
-                image_frame.image = cv::Mat(ptr_grab_result->GetHeight(),
-                                            ptr_grab_result->GetWidth(),
-                                            CV_8UC3,
-                                            (uint8_t*)pylon_image_.GetBuffer());
-            }
-            else
-            {
-                throw std::runtime_error(
-                    "Failed to access images from the camera.");
-            }
+        if (ptr_grab_result->GrabSucceeded())
+        {
+            format_converter_.Convert(pylon_image_, ptr_grab_result);
+            image_frame.image = cv::Mat(ptr_grab_result->GetHeight(),
+                                        ptr_grab_result->GetWidth(),
+                                        CV_8UC3,
+                                        (uint8_t*)pylon_image_.GetBuffer());
         }
-        catch (std::runtime_error& e)
+        else
         {
-            std::cerr << e.what() << std::endl;
+            Pylon::PylonTerminate();
```

Comment:
I am not sure if terminating here (inside `get_observation`) is good.  It might be that there was just an issue with this frame, so the backend could just skip it an continue trying to get observations.  However, when Pylon is terminated here, the camera will probably stop working completely.  So I think it is better to rely on the backend to catch this exception and not terminate here.

Reply:
Ah yes, you're right. I had not thought about that. Will fix this, thank you :)